model/project_aliases.go

```diff
@@ -585,13 +595,45 @@ func isValidRegexOrTag(curItem string, curTags, aliasTags []string, aliasRegex *
 		return true
 	}
 	for _, tag := range aliasTags {
-		if utility.StringSliceContains(curTags, tag) {
-			return true
+		// For this tag selector to match the current item's tags, it has to
+		// satisfy all the space-separated criteria defined in the selector.
+		// Note that this is functionally equivalent to project tag selector
+		// syntax, except that alias tags are not prefixed by a period.
+
+		criteria := strings.Fields(tag)
+		if len(criteria) == 0 {
+			continue
+		}
+
+		allCriteriaSatisfied := true
+		for _, criterion := range criteria {
+			if len(criterion) == 0 {
+				continue
+			}
+
+			tagName := criterion
+			var negated bool
+			if criterion[0] == '!' {
+				negated = true
+				tagName = criterion[1:]
+			}
+			if negated && utility.StringSliceContains(curTags, tagName) {
+				// Negation - tag matches the item's tags.
+				allCriteriaSatisfied = false
+				break
+			}
+			if !negated && !utility.StringSliceContains(curTags, tagName) {
+				// Intersection - tag does not match the item's tags.
+				allCriteriaSatisfied = false
+				break
+			}
 		}
-		// a negated tag
-		if len(tag) > 0 && tag[0] == '!' && !utility.StringSliceContains(curTags, tag[1:]) {
-			return true
+
+		if !allCriteriaSatisfied {
```

Comment:
readability nit: could we just say if allCriteriaSatisifed return true since we'll be continuing anyway?